zcash_keys/src/keys.rs

```diff
@@ -782,30 +833,33 @@ impl UnifiedFullViewingKey {
         &self,
         mut j: DiversifierIndex,
         request: UnifiedAddressRequest,
-    ) -> Option<(UnifiedAddress, DiversifierIndex)> {
+    ) -> Result<(UnifiedAddress, DiversifierIndex), AddressGenerationError> {
         // If we need to generate a transparent receiver, check that the user has not
         // specified an invalid transparent child index, from which we can never search to
         // find a valid index.
         #[cfg(feature = "transparent-inputs")]
-        if self.transparent.is_some() && to_transparent_child_index(j).is_none() {
-            return None;
+        if request.has_p2pkh
```

Comment:
Document this fix "can now find an address for a `j` outside the valid transparent range if you aren't requesting a transparent receiver".